utm.yaml

```diff
@@ -2429,6 +2501,111 @@ paths:
                 $ref: '#/components/schemas/ErrorResponse'
           description: The client issued too many requests in a short period of time.
 
+  #
+  # DSS Endpoint: USS AVAILABILITY
+  #
+
+  /dss/v1/uss_availability/{uss_id}:
+    summary: Availability status of a USS
+
+    put:
+      security:
+        - Authority:
+            - utm.availability_arbitration
+      summary: Set availability status of a USS.
+      operationId: setUssAvailability
+      description: Set availability status of a USS.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SetUssAvailabilityStatusParameters'
+        required: true
+      tags:
+        - Availability
+        - dss
+      responses:
+        '200':
+          description: Availability status of specified USS was successfully updated.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UssAvailabilityStatusResponse'
+        '400':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+          description: |-
+            * One or more parameters were missing or invalid.
+        '401':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+          description: Bearer access token was not provided in Authorization header,
+            token could not be decoded, or token was invalid.
+        '403':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+          description: |-
+            * The access token was decoded successfully but did not include a scope appropriate to this endpoint or the request.
+        '429':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+          description: The client issued too many requests in a short period of time.
+
+    get:
+      security:
+        - Authority:
+            - utm.availability_arbitration
+        - Authority:
+            - utm.strategic_coordination
+      summary: Get availability status of a USS.
+      operationId: getUssAvailability
+      description: Get availability status of a USS.
+      tags:
+        - Availability
+        - dss
+      responses:
+        '200':
+          description: Availability status of specified USS was successfully retrieved.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UssAvailabilityStatusResponse'
+        '400':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+          description: |-
+            * One or more parameters were missing or invalid.
```

Comment:
Should we have a specific response code if the {uss_id} is not found, maybe 404? That seems to me a likely response with a specific meaning.

Reply:
My assumption here was that all USSs would be presumed Normal unless otherwise specified by the availability arbiter, so the DSS should indicate Normal for a {uss_id} it doesn't recognize rather than 404.

Reply:
That seems a little weird since you are requesting status about an entity/endpoint that doesn't exist. Getting back a Normal status for a non-existent USS seems like it could cause problems. Even a 400 return would be ok.

Reply:
The issue there is that the DSS does not possess a list of valid USSs; it is a feature of the OAuth token system that user rights and management are handled entirely separate from the DSS.  What we're really storing here is "notices of reduced functionality" -- we could make a "/reduced_functionality/{uss_id}" endpoint instead which returns 404 if there is no known reduced functionality for the requested USS, but it seems like the current scheme is more client-friendly for clients who do not know how the API is implemented in the DSS.  I'm expecting that there will be some nominal conditions under which a USS will want to check the DSS's belief about their functionality status, and I'm a little uneasy having a 404 in a nominal workflow.

Reply:
I'm confused as to what the uss_id value represents. Is it the "owner" value as defined in the OperationReference model? I don't see it specifically defined in the yaml. If it is the owner, how does the availability_arbitration entity know the uss_id if there is not a list of valid USSs (as you indicated above)? Is the expectation that the arbitrator will be periodically checking USS status and storing a list of uss_ids?

Reply:
I'll try to add some wording for clarity.  There is only one way to identify a USS, and that's by the ID that is specified in `sub` in the access tokens that USS provides.  That ID is stored (by the DSS) in `owner` when Entities & Subscriptions are created.  The purpose of the availability arbiter is to tell the DSS when a USS (identified uniquely by this kind of ID) is down.  How the arbiter discovers the existence of USSs and how it evaluates whether those USSs are down is beyond the scope of the standard, but the example implementation illustrated in the [sequence diagram](https://github.com/astm-utm/Protocol/blob/resolve-issues-0114/diagrams/planning_with_down_uss.md) is that another USS reports the down USS (specifying the down USS's ID and the Operation they couldn't access) to the arbiter, so the arbiter gets the down USS's ID from the reporting USS.  The reporting USS knows the down USS's ID because the `owner` field is populated in the OperationReference the reporting USS receives from the DSS, which is what prompts the reporting USS's failed attempt to retrieve details for that Operation.

Another arbiter implementation could be one where the arbiter obtains a complete list of USSs (via some mechanism, but no such mechanism exists in the standard) and constantly maintains heartbeats with them, marking any USS that doesn't respond appropriately to the heartbeat as being down.  But, I don't think that implementation would be as good because it would miss the case where a USS is responding to heartbeats just fine but isn't responding appropriately to details requests.  It also seems to suboptimally allocate (heartbeat) resources as it's really not very important to discover that a USS operating only in an isolated area (or not operating at all) is down, but we want comprehensive and prompt checking of a USS interacting with other USSs.  Still, either implementation (and many more) would be enabled by the standard using the generic concept of a USS with availability arbitration scope.

Reply:
Thanks Ben for the details. Adding the uss_id parameter tag will address my initial question and your explanation of downed USS discovery/identification methods sounds good.

Reply:
I added 0fea64a476ed92243d6eea9293eb37d7cc569ce1 to address Ted's original issue.